registration/views/printing.py

```diff
@@ -16,16 +31,135 @@ def printNametag(request):
+        data_obj = signer.unsign_object(data, max_age=60)
+    except:
+        return JsonResponse({"success": False, "reason": "Invalid data"})
+
+    badge_ids = data_obj.get("badge_ids", [])
+    queryset = Badge.objects.filter(id__in=badge_ids)
+
+    badge_groups = {}
+    badge_templates = {}
+
+    for badge in queryset:
+        level = badge.effectiveLevel()
+        if not level or level == Badge.UNPAID:
+            messages.warning(
+                request, f"skipped printing {badge} because level is {level}"
+            )
+            continue
+
+        badge_template = badge.event.defaultBadgeTemplate
+        if badge_template.id not in badge_templates:
+            badge_groups[badge_template.id] = []
+            badge_templates[badge_template.id] = (badge_template, Template(badge_template.template))
+
+        level = str(level)
+        if Staff.objects.filter(
+            attendee=badge.attendee, event=badge.event
+        ).exists():
+            level = "Staff"
+        elif Dealer.objects.filter(
+            attendee=badge.attendee, event=badge.event
+        ).exists():
+            level = "Dealer"
+
+        badge_groups[badge_template.id].append({
+            "name": badge.badgeName,
+            "level": level,
+            "number": badge.badgeNumber,
+        })
+
+    with GotenbergClient(settings.GOTENBERG_HOST) as client:
+        pdfs = []
+
+        for (badge_template_id, badges) in badge_groups.items():
+            context = Context({"badges": badges})
+            (badge_template, template) = badge_templates[badge_template_id]
+            rendered = str(template.render(context))
+
+            with client.chromium.html_to_pdf() as route:
+                response = (
+                    route.size(
+                        PageSize(badge_template.paperWidth, badge_template.paperHeight)
+                    )
+                    .margins(
+                        PageMarginsType(
+                            MarginType(badge_template.marginTop),
+                            MarginType(badge_template.marginBottom),
+                            MarginType(badge_template.marginLeft),
+                            MarginType(badge_template.marginRight),
+                        )
+                    )
+                    .orient(
+                        PageOrientation(
+                            PageOrientation.Landscape
+                            if badge_template.landscape
+                            else PageOrientation.Portrait
+                        )
+                    )
+                    .scale(badge_template.scale)
+                    .string_resource(rendered, "index.html")
+                    .render_expr("window.badgeReady === true")
+                    .run()
+                )
+                pdfs.append(response.content)
+
+        finalPdf = None
+
+        if len(pdfs) == 0:
+            return JsonResponse({"success": False, "reason": "No PDFs were generated"})
```

Comment:
It might make sense to be noisier here since the end result is unexpected so we get alerted that PDF generation has failed, perhaps using Sentry's [capture_message](https://docs.sentry.io/product/sentry-basics/integrate-backend/capturing-errors/#capture-message) method.